Jenkinsfile

```diff
@@ -0,0 +1,137 @@
+                    sh """
+                        cd ${srcDir}
+                        make install
+                        cp -r ./vendor/* /go/src/
+                    """
+                }
+
+                stage('Run Tests') {
+                    sh """
+                        cd ${srcDir}
+                        make test
+                    """
+                }
+
+                if (utils.isCI()) {
+                    stage('CI: Publish Dev Image') {
+                        dockerImageVersion = stakaterCommands.getBranchedVersion("${env.BUILD_NUMBER}")
+                        
+                        sh """
+                            cd ${srcDir}
+                            export DOCKER_IMAGE=${dockerImage}
+                            export DOCKER_TAG=${dockerImageVersion}
+                            make binary-image
+                            make push
+                        """
+                    }
+                    
+                    stage('Notify') {
+                        def dockerImageWithTag = "${dockerImage}:${dockerImageVersion}"
+                        slack.sendDefaultSuccessNotification(slackWebHookURL, slackChannel, [slack.createDockerImageField(dockerImageWithTag)])
+
+                        def commentMessage = "Image is available for testing. `docker pull ${dockerImageWithTag}`"
+                        git.addCommentToPullRequest(commentMessage)
+                    }
+                } else if (utils.isCD()) {
+                    stage('CD: Tag and Push') {
+                        print "Generating New Version"
+                        def version = common.shOutput("jx-release-version --gh-owner=${repoOwner} --gh-repository=${repoName}")
+                        dockerImageVersion = version
+                        //TODO: jx-release-version won't read bumped version from .version
+                        sh """
+                            echo "${version}" > .version
+                        """
+
+                        // Render chart from templates
+                        templates.renderChart(chartTemplatesDir, chartDir, repoName, version, dockerImage)
+                        // Generate manifests from chart
+                        templates.generateManifests(chartDir, repoName, manifestsDir)
+
+                        git.commitChanges(WORKSPACE, "Bump Version to ${version}")
+
+                        print "Pushing Tag ${version} to Git"
+                        git.createTagAndPush(WORKSPACE, version)
+                        git.createRelease(version)
+
+                        print "Pushing Tag ${version} to DockerHub"
+                        
+                        sh """
+                            cd ${srcDir}
+                            export DOCKER_IMAGE=${dockerImage}
+                            export DOCKER_TAG=${dockerImageVersion}
+                            make binary-image
+                            make push
```

Comment:
@waseem-h this push will fail as we haven't done docker login here it; better if we add "login" goal in Makefile